### PR TITLE
documentation fix regarding contour and tricontour (#9088)

### PR DESCRIPTION
**Pull request by verginer**
I have changed the doc string of `contour` and `tricontour` to address the issue raised in (#9088).
Regarding the ambiguous number of levels.

## PR Summary

The change clarifies what `N` does in the contour functions and states that it can be as high
as `N+1` which would otherwise be surprising.

Related issue: #9088

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -1672,7 +1672,9 @@ def _initialize_x_y(self, z):
           contour(Z,N)
           contour(X,Y,Z,N)
 
-        contour up to *N* automatically-chosen levels.
+        contour up to *N+1* automatically-chosen levels.
+        If exactly *N* levels are required either *V* or an
```
Comment (jklymak) [inline lib/matplotlib/contour.py]: `*V*` is described right below.  I'd favour either just the first sentence or maybe even:

`contour up to *N+1* automatically chosed contour levels (*N* intervals).`

Just to make it not seem like we are bonkers to have this convention.

**Comment by verginer** [comment]
Yes I agree it is too much information. Only the clarification that there are N+1 levels is necessary.

**Comment by jklymak** [comment]
I'll merge later so long as the tests pass (though they almost certainly will, unless some of those lines got too long for PEP8).  Thanks!

**Comment by QuLogic** [comment]
Unfortunately, they're one character too long.

**Comment by jklymak** [comment]
Thanks a lot @verginer

**Pull request merged**